source/adaptive_load/metrics_plugin_impl.cc

```diff
@@ -0,0 +1,97 @@
+#include "adaptive_load/metrics_plugin_impl.h"
+
+#include "envoy/registry/registry.h"
+
+#include "external/envoy/source/common/common/assert.h"
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "absl/strings/numbers.h"
+
+namespace Nighthawk {
+
+using ::Envoy::Protobuf::util::TimeUtil;
+
+NighthawkStatsEmulatedMetricsPlugin::NighthawkStatsEmulatedMetricsPlugin(
+    const nighthawk::client::Output& nighthawk_output) {
+  // Values that could not be determined for any reason are omitted from the map.
```

Comment:
So as I understand it, incomplete results will be handled up higher in the stack?
I wonder, should we be able to handle different causes of that distinctly? Concretely,
should metrics plugins have an opportunity to have a say about that and interpret / indicate
failure (or status in general)?

Reply:
This 0.0 in the interface never really sat well with me. I brought in Envoy::StatusOr and introduced a bunch of error handling.

This does mean metrics_plugin.h needs to change, and it's just done in this PR.

Reply:
Nice, I didn’t know about ‘StatusOr’ but that seems like a good solution 👍